<!-- MHonArc v2.1.0 -->
<!--X-Subject: [pgsql&#45;jp 5405] Re: `vacuum' returned fail. -->
<!--X-From: Katsuya Sawamura <sawamura@ssemoka.co.jp> -->
<!--X-Date: Fri, 18 Sep 1998 11:42:08 +0900 -->
<!--X-Message-Id: 19980918114208W.sawamura@ssemoka.co.jp -->
<!--X-ContentType: text/plain -->
<!--X-Reference-Id: 199809180154.KAA07277@srapc451.sra.co.jp -->
<!--X-Head-End-->
<HTML>
<HEAD>
<TITLE>[pgsql-jp 5405] Re: `vacuum' returned fail. </TITLE>
<LINK REV="made" HREF="mailto:sawamura@ssemoka.co.jp">
</HEAD>
<BODY BGCOLOR="#ffffff">
<!--X-Body-Begin-->
<!--X-User-Header-->
<!--X-User-Header-End-->
<!--X-TopPNI-->
<HR>
[<A HREF="msg00197.html">Date Prev</A>][<A HREF="msg00188.html">Date Next</A>][<A HREF="msg00197.html">Thread Prev</A>][<A HREF="msg00200.html">Thread Next</A>][<A HREF="index.html#00198">Date Index</A>][<A HREF="threads.html#00198">Thread Index</A>]
<!--X-TopPNI-End-->
<!--X-MsgBody-->
<!--X-Subject-Header-Begin-->
<H1>[pgsql-jp 5405] Re: `vacuum' returned fail.</H1>
<HR>
<!--X-Subject-Header-End-->
<UL>
<LI><em>From</em>: Katsuya Sawamura &lt;<A HREF="mailto:sawamura@ssemoka.co.jp">sawamura@ssemoka.co.jp</A>&gt;</LI>
<LI><em>Date</em>: Fri, 18 Sep 1998 11:42:08 +0900</LI>
<LI><em>References</em>: &lt;<A HREF="msg00197.html">199809180154.KAA07277@srapc451.sra.co.jp</A>&gt;</LI>
</UL>
<!--X-Head-Body-Sep-Begin-->
<HR>
<!--X-Head-Body-Sep-End-->
<!--X-Body-of-Message-->
<PRE>
沢村＠三洋シリコンと申します。

From: Tatsuo Ishii &lt;t-ishii@sra.co.jp&gt;
Subject: [pgsql-jp 5404] Re: `vacuum' returned fail. 
Date: Fri, 18 Sep 1998 10:54:07 +0900

&gt; 石井です。

早速のご回答有難うございます。

&gt; &gt;まず、その時のメッセージは、
&gt; &gt;
&gt; &gt;    NOTICE:  BlowawayRelationBuffers(wbmaginfo, 59): block 64 is referenced (pr
&gt; &gt;    ivate 0, last 0, global 1)
&gt; &gt;    FATAL 1:  VACUUM (vc_rpfheap): BlowawayRelationBuffers returned -2
&gt; &gt;
&gt; &gt;となっておりました。
&gt; 
&gt; 一応ソースを見てみたのですが、残念ながら良く分かりません。エラーの直接
&gt; の原因は、vacuum が不要になったレコードを削除しようとしたが、該当レコー
&gt; ドに対応する shared memory 上のバッファが他から参照されていて(pinned)
&gt; 解放できないためです。問題はなぜこうなったかですが、そこまでは分かりま
&gt; せん。

なるほど、そう言う状態な分けですね。

&gt; お使いのバージョンが分かりませんが、直接関係するあたりのソースは、
&gt; 6.3.2 と 6.4βでは変わっていないようです。
&gt; 
&gt; というわけで、アドバイスになっていないのですが...

いえいえ、自分ではどういう状態になっているのかも分からなかったので
非常に有力なアドバイスです。

で、異常終了している状況が掴めれば、対応は簡単で(ホンマかい！？)
バックエンド (vacuum？) が、

    『shared memory 上のバッファが他から参照されている！』

と言っている (もしかしたら、勘違いしている？) のなら再起動しか
ないでしょう！と言う分けで、postmaster を立ち上げ直しました。

そして、再度問題のテーブルで vacuum を実行したところ、今度は正常に
終了しました。その後何度か繰り返してみましたが、全く問題ないです。

本当に助かりました。有難うございます。

しかし、石井さんのおかげで状況は改善されましたが、
一番肝心な『なぜそうなったのか？』を解決できないことが、
皆様に大変申し訳なく思います。
</PRE>

<!--X-Body-of-Message-End-->
<!--X-MsgBody-End-->
<!--X-Follow-Ups-->
<HR>
<UL><LI><STRONG>Follow-Ups</STRONG>:
<UL>
<LI><STRONG><A NAME="00200" HREF="msg00200.html">[pgsql-jp 5407] Re: `vacuum' returned fail.</A></STRONG>
<UL><LI><EM>From:</EM> Tatsuo Ishii &lt;t-ishii@sra.co.jp&gt;</LI></UL></LI>
</UL></LI></UL>
<!--X-Follow-Ups-End-->
<!--X-References-->
<UL><LI><STRONG>References</STRONG>:
<UL>
<LI><STRONG><A NAME="00197" HREF="msg00197.html">[pgsql-jp 5404] Re: `vacuum' returned fail.</A></STRONG>
<UL><LI><EM>From:</EM> Tatsuo Ishii &lt;t-ishii@sra.co.jp&gt;</LI></UL></LI>
</UL></LI></UL>
<!--X-References-End-->
<!--X-BotPNI-->
<UL>
<LI>Prev by Date:
<STRONG><A HREF="msg00197.html">[pgsql-jp 5404] Re: `vacuum' returned fail.</A></STRONG>
</LI>
<LI>Next by Date:
<STRONG><A HREF="msg00188.html">[pgsql-jp 5395] Re: Readline-JP</A></STRONG>
</LI>
<LI>Prev by thread:
<STRONG><A HREF="msg00197.html">[pgsql-jp 5404] Re: `vacuum' returned fail.</A></STRONG>
</LI>
<LI>Next by thread:
<STRONG><A HREF="msg00200.html">[pgsql-jp 5407] Re: `vacuum' returned fail.</A></STRONG>
</LI>
<LI>Index(es):
<UL>
<LI><A HREF="index.html#00198"><STRONG>Date</STRONG></A></LI>
<LI><A HREF="threads.html#00198"><STRONG>Thread</STRONG></A></LI>
</UL>
</LI>
</UL>

<!--X-BotPNI-End-->
<!--X-User-Footer-->
<!--X-User-Footer-End-->
</BODY>
</HTML>
